### ISSUE TITLE
Allow lowercase naming, for example `ck` instead of `CK`

### ISSUE THREAD

**Issue by warheat1990** [efcore/EFCore.CheckConstraints#77]
I'm using postgresql and usually everything is in lower case.

**Comment by roji** [comment]
@warheat1990 are you looking for https://github.com/efcore/EFCore.NamingConventions?

**Comment by warheat1990** [comment]
@roji what I mean is, by default `EFCore.CheckConstraints` hardcoded the check constrains as `CK_{tableName}_{columnName}_Enum', it would be nice if we have option to make it lower case like `ck_{tablename}_{columnname}_enum` as postgresql usually use `snake_case` naming.

**Comment by roji** [comment]
@warheat1990 are you saying that when using both EFCore.CheckConstraints *and* EFCore.NamingConventions, you're still seeing uppercase CK instead of the expected lowercase?

**Comment by warheat1990** [comment]
@roji that's correct, but I'm actually not aware that it's supposed to respect the setting on NamingConvention because I have a quick glance at the code and it seems hardcoded to CK_tablename_columnname_Enum

**Comment by roji** [comment]
@warheat1990 the point is that the value "hard-coded" by @warheat1990 should get processed by the naming convention plugin, and made lower-case at that point. The EFCore.CheckConstraint plugin shouldn't start dealing with naming convention and casing issues on its own, else we start duplicating EFCore.NamingConventions into it.

So this is likely an issue with ordering/interaction of the two conventions coming from the two plugins - I'll try to take a look soon.

**Comment by warheat1990** [comment]
Here's the code if it helps

```
builder.Services.AddDbContext<ApplicationContext>(
    options => options
        .UseNpgsql(configuration["DefaultConnection"])
        .UseSnakeCaseNamingConvention()
        .UseAllCheckConstraints()
);
```

![Untitled](https://user-images.githubusercontent.com/4485011/170810139-cef673f8-43ca-4781-b9b8-202a9cf930b6.png)

**Comment by cremor** [comment]
@roji Any news on this? I have the same problem.

**Comment by roji** [comment]
@cremor no news. I don't expect to have any time to look at it, and in any case this likely requires some infrastructure around EF conventions that isn't there yet, i.e. convention layering (/cc @andriysvyryd).
